requirements-dev.txt

```diff
@@ -22,7 +22,7 @@ requirements-parser
 black==23.9.1
 
 # Required for test
-pytest==7.2.0
+pytest==7.4.2
```

Comment:
Do we need to pin `pytest` at all? I know we had a reason at some point, but that was a few years ago.... and may just have been to make it play nice with VSCode back then.

Reply:
OK, but see the comment about the 'release' Action in GitHub

Reply:
Yes, I addressed that one just now. Please take a look to see if it makes sense to you. I rather like the solution of having just 1 place to specify the versions 😄

Reply:
Ideally we will converge on a single source of truth....

For those others (such as myself) unaware of this `pip` feature:
https://pip.pypa.io/en/stable/user_guide/#constraints-files